spec/lib/api/2.0/tags-spec.js

```diff
@@ -154,5 +157,108 @@ describe('Http.Api.Tags', function () {
+                        expect(nodesApi.getNodesByTag).to.be.calledWith('tag001');
+                        expect(nodesApi.setNodeWorkflow).to.be.calledTwice;
+                        expect(nodesApi.setNodeWorkflow.firstCall).to.be.calledWith(
+                            {name: 'Graph.test', domain: 'test', options: { test: true } },
+                            'nodeId1' );
+                        expect(nodesApi.setNodeWorkflow.secondCall).to.be.calledWith(
+                            {name: 'Graph.test', domain: 'test', options: { test: true } },
+                            'nodeId2' );
+                    });
+            });
+
+            it('should support specify workflow name via query', function() {
+                sandbox.stub(nodesApi, 'getNodesByTag').withArgs('tag001').resolves([
+                    { id: 'nodeId1' },
+                    { id: 'nodeId2' }
+                ]);
+                sandbox.stub(nodesApi, 'setNodeWorkflow')
+                    .onFirstCall().resolves({ id: 'graphId1' })
+                    .onSecondCall().resolves({ id: 'graphId2' });
+                return helper.request()
+                    .post('/api/2.0/tags/tag001/nodes/workflows?name=Graph.test')
+                    .send({ domain: 'test',  options: { test: true} })
+                    .expect('Content-Type', /^application\/json/)
+                    .expect(202, [ {id: 'graphId1'}, {id: 'graphId2'}])
+                    .then(function() {
+                        expect(nodesApi.getNodesByTag).to.be.calledWith('tag001');
+                        expect(nodesApi.setNodeWorkflow).to.be.calledTwice;
+                        expect(nodesApi.setNodeWorkflow.firstCall).to.be.calledWith(
+                            {name: 'Graph.test', domain: 'test', options: { test: true } },
+                            'nodeId1' );
+                        expect(nodesApi.setNodeWorkflow.secondCall).to.be.calledWith(
+                            {name: 'Graph.test', domain: 'test', options: { test: true } },
+                            'nodeId2' );
+                    });
+            });
+
+            it('should succeed if no nodes are binded with tag', function() {
+                sandbox.stub(nodesApi, 'getNodesByTag').withArgs('tag001').resolves([]);
+                sandbox.stub(nodesApi, 'setNodeWorkflow').resolves({id: 'graphId'});
+                return helper.request()
+                    .post('/api/2.0/tags/tag001/nodes/workflows')
+                    .send({ name: 'Graph.test', options: { test: true} })
+                    .expect('Content-Type', /^application\/json/)
+                    .expect(202)
+                    .then(function() {
+                        expect(nodesApi.getNodesByTag).to.be.calledWith('tag001');
+                        expect(nodesApi.setNodeWorkflow).to.not.be.called;
+                    });
+            });
+
+            it('should 400 if tag name is not existing', function() {
```

Comment:
Actually, this is somewhat beyond the scope for the PR.

The response status depends on what's kind of Error  the `nodesApi.getNodesByTag` returns, if it is returns `NotFoundError`, then the error code is 404, otherwise is 400. If you look at the implementation of `nodesApi.getNodesByTag`, actually it never reports `NotFoundError` if tag name is not existing, instead, it returns empty array.

Since a lot other APIs call `nodesApi.getNodesByTag` as well, sometimes expect empty array, sometimes expect `NotFoundError`, so I'd suggest fixing it in another PR.

At least for this PR, I can change to stub the `nodesApi.getNodesByTag` to return `NotFoundError` and expect 404.